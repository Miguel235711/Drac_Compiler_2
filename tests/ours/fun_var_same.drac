var x;

fun(y){
    x = 3;
}

main(){
    main2 = 3;
    main2();
}
var main;
main(){ -- error, should fail because of redefinition
    var main2,main2; -- 2Â° main2 should fail because of redefinition
}
var main2;
main2(){
    var main2;
}
var main2;-- error, should fail because of redefinition