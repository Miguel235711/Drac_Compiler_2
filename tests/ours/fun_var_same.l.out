.\tests\ours\fun_var_same.drac
label: var content: var
label: <id> content: x
label: ; content: ;
label: <id> content: fun
label: ( content: (
label: <id> content: y
label: ) content: )
label: { content: {
label: <id> content: x
label: = content: =
label: <lit-int> content: 3
label: ; content: ;
label: } content: }
label: <id> content: main
label: ( content: (
label: ) content: )
label: { content: {
label: <id> content: main2
label: = content: =
label: <lit-int> content: 3
label: ; content: ;
label: <id> content: main2
label: ( content: (
label: ) content: )
label: ; content: ;
label: } content: }
label: var content: var
label: <id> content: main
label: ; content: ;
label: <id> content: main
label: ( content: (
label: ) content: )
label: { content: {
label: var content: var
label: <id> content: main2
label: , content: ,
label: <id> content: main2
label: ; content: ;
label: } content: }
label: var content: var
label: <id> content: main2
label: ; content: ;
label: <id> content: main2
label: ( content: (
label: ) content: )
label: { content: {
label: var content: var
label: <id> content: main2
label: ; content: ;
label: } content: }
label: var content: var
label: <id> content: main2
label: ; content: ;
